src/nurse_ussd.js

```diff
@@ -175,558 +152,359 @@ go.app = function() {
                 ],
-
                 next: function(choice) {
-                    self.contact.extra.id_type = choice.value;
-
-                    return self.im.contacts
-                        .save(self.contact)
-                        .then(function() {
-                            return {
-                                sa_id: 'states_sa_id',
-                                passport: 'states_passport_origin',
-                                none: 'states_birth_year'
-                            } [choice.value];
-                        });
+                    return choice.value;
                 }
             });
         });
 
-        self.add('states_sa_id', function(name, opts) {
-            var error = $('Sorry, the mother\'s ID number did not validate. ' +
-                          'Please reenter the SA ID number:');
-
-            var question = $('Please enter the pregnant mother\'s SA ID ' +
-                            'number:');
-
+        self.add('st_faccode', function(name) {
+            var owner = self.user.extra.working_on === "" ? 'your' : 'their';
+            var error = $("Sorry, that code is not recognized. Please enter the 6-digit facility code again, e. 535970:");
+            var question = $("Please enter {{owner}} 6-digit facility code:")
+                .context({owner: owner});
             return new FreeText(name, {
                 question: question,
-
                 check: function(content) {
-                    if (!go.utils.validate_id_sa(content)) {
-                        return error;
-                    }
+                    return go.utils
+                        .validate_clinic_code(self.im, content.trim())
+                        .then(function(facname) {
+                            if (!facname) {
+                                return error;
+                            } else {
+                                self.contact.extra.facname = facname;
+                                return self.im.contacts
+                                    .save(self.contact)
+                                    .then(function() {
+                                        return null;  // vumi expects null or undefined if check passes
+                                    });
+                            }
+                        });
                 },
-
                 next: function(content) {
-                    self.contact.extra.sa_id = content;
-
-                    var id_date_of_birth = go.utils.extract_id_dob(content);
-                    self.contact.extra.birth_year = moment(id_date_of_birth, 'YYYY-MM-DD').format('YYYY');
-                    self.contact.extra.birth_month = moment(id_date_of_birth, 'YYYY-MM-DD').format('MM');
-                    self.contact.extra.birth_day = moment(id_date_of_birth, 'YYYY-MM-DD').format('DD');
-                    self.contact.extra.dob = id_date_of_birth;
-
+                    self.contact.extra.faccode = content.trim();
```

Comment:
Probably because I wrote the `next` first and added validation later and overlooked the optimization opportunity. Fixed now.